src/arch/riscv/pmp.cc

```diff
@@ -0,0 +1,153 @@
+#include "arch/riscv/pmp.hh"
+
+#include "arch/generic/tlb.hh"
+#include "base/addr_range.hh"
+#include "base/types.hh"
+#include "math.h"
+#include "mem/request.hh"
+#include "params/PMP.hh"
+#include "sim/sim_object.hh"
+
+PMP::PMP(const Params &params) :
+SimObject(params), Max_Entries(params.max_pmp)
+{
+
+ for (int i=0; i<Max_Entries; i++){
+
+    pmp_entry* entry = new pmp_entry;
+    pmp_table.push_back(entry);
```

Comment:
Use `emplace` instead and don't store pointers.

Reply:
done